src/cart/crt_hg_proc.c

```diff
@@ -208,17 +208,9 @@ crt_proc_d_rank_list_t(crt_proc_t proc, d_rank_list_t **data)
 			D_GOTO(out, rc = 0);
 		}
 
-		D_ALLOC_PTR(rank_list);
-		if (rank_list == NULL)
-			D_GOTO(out, rc = -DER_NOMEM);
-		D_ALLOC_ARRAY(rank_list->rl_ranks, nr);
-		if (rank_list->rl_ranks == NULL) {
-			D_FREE(rank_list);
-			D_GOTO(out, rc = -DER_NOMEM);
-		}
+		rank_list = d_rank_list_alloc(nr);
```

Comment:
Good spot, but you need to check rank_list against NULL.

Reply:
Thanks. Yes, I removed a bit more. :-) Thanks to catch this.